waku/waku_archive/archive.nim

```diff
@@ -29,28 +22,43 @@ logScope:
   topics = "waku archive"
 
 const
-  DefaultPageSize*: uint = 20
-  MaxPageSize*: uint = 100
+  #TODO if waku message are 150KiB max are those numbers still appropriate???
+  DefaultPageSize*: uint = 20 # 2.93 MiB
+  MaxPageSize*: uint = 100 # 14.65 MiB
 
-## Message validation
+# Retention policy
+const WakuArchiveDefaultRetentionPolicyInterval* = chronos.minutes(30)
+
+# Metrics reporting
+const WakuArchiveDefaultMetricsReportInterval* = chronos.minutes(1)
+
+# Message validation
+# 20 seconds maximum allowable sender timestamp "drift"
+const MaxMessageTimestampVariance* = getNanoSecondTime(20) 
 
-type
-  MessageValidator* = ref object of RootObj
+## Archive
 
-  ValidationResult* = Result[void, string]
+type WakuArchive* = ref object
+  started: bool
 
-method validate*(validator: MessageValidator, msg: WakuMessage): ValidationResult {.base.} = discard
+  driver: ArchiveDriver
 
-# Default message validator
+  retentionPolicy: Option[RetentionPolicy]
 
-const MaxMessageTimestampVariance* = getNanoSecondTime(20) # 20 seconds maximum allowable sender timestamp "drift"
+proc new*(T: type WakuArchive,
+          driver: ArchiveDriver,
+          retentionPolicy = none(RetentionPolicy)):
+          Result[T, string] =
 
-type DefaultMessageValidator* = ref object of MessageValidator
+  let archive =
+    WakuArchive(
+      driver: driver,
+      retentionPolicy: retentionPolicy,
+    )
 
-method validate*(validator: DefaultMessageValidator, msg: WakuMessage): ValidationResult =
-  if msg.timestamp == 0:
-    return ok()
```

Comment:
Is timestamp becoming mandatory in the new Store protocol version (v3.) Sorry I couldn't fine the new Store definition

Reply:
Yes this is part of the Store v3 specification https://github.com/vacp2p/rfc/tree/docs/new-store-v3-protocol/content/docs/rfcs/13#waku-message-store-eligibility